python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py

```diff
@@ -0,0 +1,448 @@
+    And: KmsKeyId parameter is configured
+    And: KmsKeyId key on DescribeFileSystems is not matching KmsKeyId parameter (or not present)
+   Then: Return NON_COMPLIANT on this EFS Filesystem
+
+  Scenario 5:
+  Given: At least one EFS filesystem is present
+    And: The "Encrypted" key is set to True on DescribeFileSystems
+    And: KmsKeyId parameter is configured
+    And: KmsKeyId key on DescribeFileSystems is matching KmsKeyId parameter
+   Then: Return COMPLIANT on this EFS Filesystem
+"""
+
+
+import json
+import datetime
+import boto3
+import botocore
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::EFS::FileSystem'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+#############
+# Main Code #
+#############
+
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+
+    efs_client = get_client('efs', event)
+
+    # get all the file systems
+    all_file_systems = get_all_file_systems(efs_client)
+
+    evaluations = []
+
+    # check whether atlease one file system exist
+    if not all_file_systems:
+        return build_evaluation(event['accountId'], 'NOT_APPLICABLE', event)
+
+    for each_efs in all_file_systems:
+        # check if file system is encrypted
+        if each_efs['Encrypted']:
+            # check if any valid paraometer is provided
+            if valid_rule_parameters:
+                # if valid parameter is provided then compare parameter with KmsKeyId
+                if each_efs['KmsKeyId'] == valid_rule_parameters:
+                    evaluations.append(build_evaluation(each_efs['FileSystemId'], 'COMPLIANT', event))
+                else:
+                    evaluations.append(build_evaluation(each_efs['FileSystemId'], 'NON_COMPLIANT', event, annotation='EFS is encrypted but KeyId is not matching the paramter supplied'))
+            else:
+                evaluations.append(build_evaluation(each_efs['FileSystemId'], 'COMPLIANT', event))
+        else:
+            evaluations.append(build_evaluation(each_efs['FileSystemId'], 'NON_COMPLIANT', event, annotation='EFS is is not encryprted'))
+
+    return evaluations
+
+
+def get_all_file_systems(efs_client):
+    all_file_systems = []
+
+    file_systems = efs_client.describe_file_systems(MaxItems=15)
+    all_file_systems += file_systems['FileSystems']
+
+    while True:
+        if "NextMarker" in file_systems:
+            file_systems = efs_client.describe_file_systems(Marker=file_systems['NextMarker'], MaxItems=1000)
+            all_file_systems += file_systems['FileSystems']
+        else:
+            break
+
+    return all_file_systems
+
+
+def evaluate_parameters(rule_parameters):
+
+    # If parameter is given check whether it's ARN, else ignore it as it is optional paramter.
+    if 'KmsKeyId' not in rule_parameters:
+        return False

```

Comment:
I prefer to keep the output of functions constant. Always return a dict in this case.

Start with 
```python
if 'KmsKeyId' not in rule_parameters:
        return {}